includes/general.php

```diff
@@ -1543,97 +1543,152 @@ function pods_by_title( $title, $output = OBJECT, $type = 'page', $status = null
+ * @param string|array $name   The field name, or an associative array of parameters.
+ * @param boolean      $single For tableless fields, to return the whole array or the just the first item.
  *
- * @return mixed Field value
+ * @return mixed Field value.
  *
  * @since 2.1.0
  */
-function pods_field_display( $pod, $id = false, $name = null, $single = false ) {
+function pods_field_display( $pod, $id = null, $name = null, $single = false ) {
 
 	// allow for pods_field_display( 'field_name' );
 	if ( null === $name ) {
 		$name   = $pod;
 		$single = (boolean) $id;
 
+		$pod = null;
+		$id  = null;
+	}
+
+	if ( null === $pod && null === $id ) {
 		$pod = get_post_type();
 		$id  = get_the_ID();
 	}
 
-	$pod = pods( $pod, $id );
+	$pod_object = pods( $pod, $id );
 
-	if ( is_object( $pod ) ) {
-		return $pod->display( $name, $single );
+	if ( is_object( $pod_object ) && $pod_object->exists() ) {
+		return $pod_object->display( $name, $single );
 	}
 
 	return null;
 }
 
 /**
- * Get a field raw value from a Pod
+ * Get a field raw value from a Pod.
  *
- * @param string       $pod    The pod name
- * @param mixed        $id     (optional) The ID or slug, to load a single record; Provide array of $params to run
- *                             'find'
- * @param string|array $name   The field name, or an associative array of parameters
- * @param boolean      $single (optional) For tableless fields, to return the whole array or the just the first item
+ * @param string|null  $pod    The pod name.
+ * @param mixed|null   $id     The ID or slug of the item.
+ * @param string|array $name   The field name, or an associative array of parameters.
+ * @param boolean      $single For tableless fields, to return the whole array or the just the first item.
  *
- * @return mixed Field value
+ * @return mixed Field value.
  *
  * @since 2.1.0
  */
-function pods_field_raw( $pod, $id = false, $name = null, $single = false ) {
+function pods_field_raw( $pod, $id = null, $name = null, $single = false ) {
 
 	// allow for pods_field_raw( 'field_name' );
 	if ( null === $name ) {
 		$name   = $pod;
 		$single = (boolean) $id;
 
+		$pod = null;
+		$id  = null;
+	}
+
+	if ( null === $pod && null === $id ) {
 		$pod = get_post_type();
 		$id  = get_the_ID();
 	}
 
-	return pods( $pod, $id )->raw( $name, $single );
+	$pod_object = pods( $pod, $id );
+
+	if ( is_object( $pod_object ) && $pod_object->exists() ) {
+		return $pod_object->raw( $name, $single );
+	}
+
+	return null;
+
+}
 
+/**
+ * Update a field value for a Pod.
+ *
+ * @param string|null  $pod   The pod name.
+ * @param mixed|null   $id    The ID or slug of the item.
+ * @param string|array $name  The field name, or an associative array of parameters.
+ * @param boolean      $value Value to save.
+ *
+ * @return int|false The item ID or false if not saved.
+ *
+ * @since 2.7.2
+ */
+function pods_field_update( $pod, $id = null, $name = null, $value = null ) {
+
+	// allow for pods_field( 'field_name' );
```

Comment:
* :warning: This comment is 46% valid code; is this commented out code?
* :no_entry_sign: Inline comments must end in full-stops, exclamation marks, or question marks

<sub>hash: f1cce83f0e937c3844679dc2ab1a4c3a</sub>